contrib/build-linux/appimage/build.sh

```diff
@@ -55,44 +49,31 @@ $SUDO docker build -t $IMGNAME \
     contrib/build-linux/appimage \
     || fail "Failed to create docker image"
 
-# This is the place where we checkout and put the exact revision we want to work
-# on. Docker will run mapping this directory to /opt/electrumabc
-FRESH_CLONE=`pwd`/contrib/build-linux/fresh_clone
-FRESH_CLONE_DIR=$FRESH_CLONE/$GIT_DIR_NAME
+PROJECT_ROOT=`pwd`
 MAPPED_DIR=/opt/electrumabc
 
-(
-    $SUDO rm -fr $FRESH_CLONE && \
-        mkdir -p $FRESH_CLONE && \
-        cd $FRESH_CLONE  && \
-        git clone $GIT_REPO && \
-        cd $GIT_DIR_NAME && \
-        git checkout $REV
-) || fail "Could not create a fresh clone from git"
```

Comment:
You want to keep so insurance that the tree is not dirty for running the release build, which was a side effect of the clone.
I suggest you check that before the script continues. It might be good to have an automatic clean (like `git -xffd`) with an option to disable it (useful to run tests)